daemon/daemon_main.go

```diff
@@ -1328,6 +1331,24 @@ func runDaemon() {
 		}()
 	}
 
+	// This must be before the API is served to prevent endpoints from being
+	// created before we migrate to the new ENI datapath. It also must be
+	// before the health initialization logic which might create the health
+	// endpoint, for the same reasons as being before the API is served. We
+	// want to ensure that this migration logic runs before any endpoint is
+	// being created.
+	if option.Config.IPAM == option.IPAMENI {
+		err, revErr := linuxrouting.MigrateENIDatapath(&interfaceNumberGetter{})
+		switch {
+		case err != nil && revErr != nil:
+			log.WithError(err).Fatal("Failed to migrate ENI datapath. The original datapath failed to be reinstated. Connectivity is not guaranteed.")
```

Comment:
I'm not exactly sure what the mitigation is. This case is obviously the worst-case. We failed to migrate and we failed to reinstate the original datapath configuration. It could have failed to reinstate completely or could have failed on the last operation. Besides manual intervention, I'm not sure what we can recommend. Thoughts?

Reply:
I'd say the log message should include pointers how to do the manual intervention. Otherwise, less experienced users will have to tear down a node.